flik/linesearch/interpolations.py

```diff
@@ -0,0 +1,76 @@
+"""Interpolations for step length selection."""
+import numpy as np
+
+def quad_approx(func, grad, val, alpha, direction):
+    """
+    Quadratic Approximation
+
+    Parameters
+    ----------
+    func: callable
+        Scalar-valued function in form f(xk), where xk is is a 1-dim array
+    grad: callable
+        Vector-valued gradient function in form f'(xk), where xk is 1-dim array
+    val: np.ndarray(N,)
+        Objective function value
+    alpha: float
+        Step length
+    direction: np.ndarray(N,)
+        Descent direction
+
+    Raises
+    ------
+    TypeError
+        If func is not callable
+        If grad is not callable
+        If variable is not 1-dim array of float
+        If descent direction is not 1-dim array of float
+        If alpha is not float
+        If alpha_one is not float
+        If func does not return float
+        If grad does not return np.ndarray
+    ValueError
+        If descent direction and value is not the same size
+        If alpha is not between zero and one
+        If alpha_one is not between zero and alpha
+
+    Returns
+    -------
+    alpha_one: float
+        New step length
+    """
+    # Check parameters
+    if not callable(func):
+        raise TypeError('Gradient should be a function')
+    if not callable(grad):
+        raise TypeError('Gradient should be a function')
+    if not (isinstance(val, np.ndarray) and val.ndim == 1):
+        raise TypeError('Variable should be an 1-dim array of float')
+    if not (isinstance(direction, np.ndarray) and direction.ndim == 1):
+        raise TypeError('Descent direction should be an 1-dim array of float')
+    if val.size != direction.size:
+        raise ValueError('Size of direction and variable should be same')
+    if not isinstance(alpha, float):
+        raise TypeError('Alpha should be a float')
+    if not 0 < alpha < 1:
+        raise ValueError('Alpha should be greater than zero and smaller than one')
+    if not isinstance(func(val), float):
```

Comment:
It would be better to move the `phi_zero`'s and use `phi_zero` and `phi_zero_prime` instead.